airflow/api/common/delete_dag.py

```diff
@@ -64,18 +64,6 @@ def delete_dag(dag_id: str, keep_records_in_log: bool = True, session: Session =
     if dag is None:
         raise DagNotFound(f"Dag id {dag_id} not found")
 
-    # deleting a DAG should also delete all of its subdags
-    dags_to_delete_query = session.execute(
-        select(DagModel.dag_id).where(
-            or_(
-                DagModel.dag_id == dag_id,
-                and_(DagModel.dag_id.like(f"{dag_id}.%"), DagModel.is_subdag),
-            )
-        )
-    )
-
-    dags_to_delete = [dag_id for (dag_id,) in dags_to_delete_query]
```

Comment:
This would fail to actually delete the non-sub DAG (the `== dag_id` part)

Reply:
I am handling it at the bottom -- Line 77

Reply:
Does this not remove deletion altogether?

Reply:
seen the line 77 :)